### PR TITLE
Add a test for the powershell hang fix

### PR DESCRIPTION
**Pull request by daxian-dbw**
Add test for the fix in PR #5356 
Also add a minor change to build.psm1 to ignore the 'Sync-PSTags' warning.

### CODE REVIEW

File: test/powershell/engine/Api/BasicEngine.Tests.ps1 [hunk 1]
```diff
@@ -21,3 +21,29 @@ Describe 'Basic engine APIs' -Tags "CI" {
...
+-c $rs = [runspacefactory]::CreateRunspacePool(1,5)
+$rs.Open()
+$ps = [powershell]::Create()
+$ps.RunspacePool = $rs
+$null = $ps.AddScript(1).Invoke()
+write-host should_not_hang_at_exit
+exit
+'@
+        $process = Start-Process pwsh -ArgumentList $command -PassThru
+        for ($i = 0; -not $process.HasExited -and $i -lt 5; $i++) {
```
Comment (adityapatwardhan) [inline test/powershell/engine/Api/BasicEngine.Tests.ps1]: Please use the helper at: https://github.com/PowerShell/PowerShell/blob/master/test/tools/Modules/HelpersCommon/HelpersCommon.psm1#L1

File: test/powershell/engine/Api/BasicEngine.Tests.ps1 [hunk 2]
```diff
@@ -21,3 +21,29 @@ Describe 'Basic engine APIs' -Tags "CI" {
...
+            Start-Sleep -Seconds 1
+        }
+
+        $expect = "powershell process exits in 5 seconds"
```
Comment (adityapatwardhan) [inline test/powershell/engine/Api/BasicEngine.Tests.ps1]: Should the verification be like:
```Powershell
$process.HasExited | Should Be $true
Stop-Process $process -ErrorAction SilentlyContinue

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: test/powershell/engine/Api/BasicEngine.Tests.ps1 (hunk 1 above)
Comment (daxian-dbw) [reply test/powershell/engine/Api/BasicEngine.Tests.ps1]: OK, I will use `Wait-UntilTrue`

File: test/powershell/engine/Api/BasicEngine.Tests.ps1 (hunk 2 above)
Comment (daxian-dbw) [reply test/powershell/engine/Api/BasicEngine.Tests.ps1]: The `Should be $true` check will not generate a useful error message when it fails, so I prefer to the current check. But I will add `-ErrorAction SilentlyContinue`

**Comment by daxian-dbw** [comment]
@adityapatwardhan Your comments are addressed. Please take another look. Thanks!

**Review by adityapatwardhan — APPROVED** [review]

**Review by iSazonov — APPROVED** [review]
Leave a comment

**Review by PaulHigin — APPROVED** [review]
LGTM

**Pull request merged**